### DISCUSSION TITLE
Generate Ubuntu 18.04 | 20.04 image fails certificate error

### DISCUSSION THREAD

**Discussion by wmayvis** [actions/virtual-environments#5746] [General questions]
A couple of weeks, I can't generate new  ubuntu 1804 or ubuntu 2004 images.
The execution failes due certificate errors. See build log.

![buidlfailed](https://user-images.githubusercontent.com/7701921/173774648-cc291d66-919a-4934-85bb-8a0e3e022799.png)

**Comment by al-cheb** [comment]
Use the `main` brach to build images.

**Reply by wmayvis** [reply]
The first step I will do a git clone, so I use the main branch.
Cloning into 'virtual-environments'...
Already on 'main'
Your branch is up to date with 'origin/main'.
Already up to date.

It looks like that ftp.byfly.by an alternated host is that is hosted in Belarus and that should maybe block.

**Reply by mikhailkoliada** [reply]
I've just checked our regular builds and found nothing suspicious, all the downloading is happening using the download.opensuse.org address, so might be repo owners have some kind of load balancing/round robin, do failed builds always refer to byfly.by?

**Reply by wmayvis** [reply]
There is also a lot of certificate issues. 
Err:14 https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_20.04  crun 100:1.2-2
    azure-arm:   Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown.  Could not handshake: Error in the certificate verification. [IP: ::ffff:146.112.61.104 443]
    azure-arm: Err:22 https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_20.04  podman 100:3.4.2-5
    azure-arm:   Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown.  Could not handshake: Error in the certificate verification. [IP: ::ffff:146.112.61.104 443]
    azure-arm: Err:23 https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_20.04  skopeo 100:1.5.0-3
    azure-arm:   Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown.  Could not handshake: Error in the certificate verification. [IP: ::ffff:146.112.61.104 443]
    azure-arm: Err:25 https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_20.04  containernetworking-plugins 100:1.1.1~1
    azure-arm:   Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown.  Could not handshake: Error in the certificate verification. [IP: ::ffff:146.112.61.104 443]
    azure-arm: Fetched 13.5 MB in 10s (1287 kB/s)

**Comment by al-cheb** [comment]
Looks like firewall blocking - https://support.umbrella.com/hc/en-us/articles/115001357688-What-are-the-Cisco-Umbrella-Block-Page-IP-Addresses-

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
